### DISCUSSION TITLE
[Thesis] Blockprogrammiersprache (Programmieren für Einsteiger)

### DISCUSSION THREAD

**Discussion by AMatutat** [Programmiermethoden/Dungeon#1383] [Ideas]
Die Idee ist, eine Blockprogrammiersprache per GUI für non-programmierende SchülerInnen anzubieten, die einfache Logiken anpassen möchten. Ziel ist es nicht, komplexe Dungeon-Konstrukte zu implementieren, sondern z.B ein vorgegebenes Labyrinth-Level mit einem Helden zu gestalten. In der Blocksprache soll das Verhalten des Helden entwickelt werden, beispielsweise "Gehe 3 Schritte nach links, dann gehe 4 Schritte nach rechts, wenn eine Wand kommt, gehe nach links", usw. 

![example](https://www.softwaretestinghelp.com/wp-content/qa/uploads/2020/09/Creatio.png)

Die Logik in der Blocksprache soll dann in Echtzeit im Spiel geladen und angezeigt werden. Die Echtzeit-Integration bedeutet, dass das Spiel nicht neu gestartet werden muss, sondern dass das Level einfach neu gestartet werden kann. Die GUI soll über Knöpfe zum Starten des Helden und zum Neustarten des Levels verfügen.

Zur Lösung müssen entsprechende Szenarien im Dungeon (Level, Spielsituationen, NPC, ...) vorbereitet und vorgegeben werden, die dann über die Programmierung in der Block-Sprache "nur noch" gesteuert werden.

siehe auch #1070

**Comment by cagix** [comment]
@AMatutat wir haben blockly. inwieweit wäre das hier vom konzept her anders?
